### ISSUE TITLE
Only activate httpBL rules when requested by user

### ISSUE THREAD

**Issue by lifeforms** [SpiderLabs/owasp-modsecurity-crs#313]
During testing of CRS3 I often get a `Rule processing failed.` audit log message.

Turning up the `SecDebugLogLevel` reveals that this is due to rule 910130 (RBL lookup of .dnsbl.httpbl.org).

```
[4] Executing operator "rbl" with param "dnsbl.httpbl.org" against TX:real_ip.
[9] Target value: "192.168.0.2"
[4] RBL httpBl called but no key defined: set SecHttpBlKey
[4] Operator completed in 18 usec.
[4] Operator error: RBL httpBl called but no key defined: set SecHttpBlKey
[4] Rule returned -1.
[1] Rule processing failed.
```

It seems not a big problem, but nevertheless, getting this error in audit logs is scary and uninformative to the user.

By default, the HTTP blacklist key and settings are commented out in modsecurity_crs_10_setup.conf (`SecHttpBlKey` and rule 900025), so the HTTPBL related rules can't work anyway.

I propose that in REQUEST-10-IP-REPUTATION.conf, we **skip over** the HTTPBL related rules (910130, 910140, 910150, 910160, 910170) unless the user has enabled at least one Project Honeypot HTTP Blacklist check.

To do this, we could check for the variables `tx.block_search_ip`, `tx.block_suspicious_ip`, `tx.block_harvester_ip` and `tx.block_spammer_ip` from rule 900025.

**Comment by dune73** [comment]
This is an issue.

So you would skip if all variables are set to 0?

What about 910180? I think that can be skipped as well.

**Comment by csanders-git** [comment]
I think your solution is particularly elegant... Not sure why we were doing it this way... we even have an END_RBL check already specified so we should just add them up and see if it == 0 if it does then we skip to END_RBL? Am i understanding this correctly?

**Comment by lifeforms** [comment]
@dune73 @csanders-git Hmmm, what if we'd want to add more RBL rules later? If we skip to `END_RBL_CHECK` we are skipping them all. I think it's probably clearer if we insert a new marker `END_HTTPBL_CHECK` after 910170, and just skip to that one.

**Comment by dune73** [comment]
If we add something later, there is going to be a separate patch, where we can add END_HTTPBL_CHECK if needed. I second @csanders-git idea to use the existing END_RBL_LOOKUP or the existing END_RBL_CHECK.

**Comment by csanders-git** [comment]
yeah I don't think it's a bad idea to add another marker.

**Comment by lifeforms** [comment]
Merged as #330
